### PR TITLE
[F] Add Journals routes and containers

### PR DESCRIPTION
**Pull request by 1aurend**
Adds routes and containers for journals at varying stages of development:

`IssuesList` renders UI with mocked data.
`JournalsList` renders stringified mock data.
`JournalDetail` and `VolumeDetail` just render the container name. Redux hooks in progress.

This PR also includes a number of new hooks for repetitive container logic:

- `usePaginationState`
- `useFilterState`
- `useUrlFromState`
- and not a hook but `pageChangeHandlerCreator` helper

### CODE REVIEW

File: client/src/frontend/containers/JournalDetail/index.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+} from "hooks";
+import { RegisterBreadcrumbs } from "global/components/atomic/Breadcrumbs";
+import Layout from "frontend/components/layout";
+
+export default function JournalDetailContainer({ location, match }) {
+  const { journal, issues, issuesMeta } = useSelectJournal();
+  const settings = useSelectSettings();
+  const { paginationState, handlePageChange } = usePaginationState(location);
+  const { filterState, updateFilterState } = useFilterState(location);
+  if (!journal) return null;
```
Comment (1aurend) [inline client/src/frontend/containers/JournalDetail/index.js:22]: To Do: update to use the most recent versions of hooks

File: client/src/frontend/containers/JournalDetail/index.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    if (heroStyles?.mediumLandscape) return heroStyles.mediumLandscape;
+    return null;
+  };
+
+  const pageChangeHandlerCreator = pageParam => {
+    return event => {
+      event.preventDefault();
+      handlePageChange(pageParam);
+    };
+  };
```
Comment (1aurend) [inline client/src/frontend/containers/JournalDetail/index.js:51]: To Do: import new helper

File: client/src/frontend/containers/JournalWrapper/index.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import PropTypes from "prop-types";
+import lh from "helpers/linkHandler";
+import { useSelectSettings, useSelectJournal } from "hooks";
+import { RedirectToFirstMatch, childRoutes } from "helpers/router";
+import CheckFrontendMode from "global/containers/CheckFrontendMode";
+import EventTracker, { EVENTS } from "global/components/EventTracker";
+
+export default function JournalWrapper({ location, match, route }) {
+  const settings = useSelectSettings();
+  const { journal, journalResponse } = useSelectJournal();
```
Comment (1aurend) [inline client/src/frontend/containers/JournalWrapper/index.js:11]: To Do: add dispatch hook for journal (and its volumes) and update child props

File: client/src/frontend/containers/VolumeDetail/index.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import { RegisterBreadcrumbs } from "global/components/atomic/Breadcrumbs";
+import Layout from "frontend/components/layout";
+
+export default function VolumeDetailContainer({ location, match }) {
+  const { journal } = useSelectJournal();
+  const { volume, issues } = useSelectVolume();
+  const settings = useSelectSettings();
+  const { paginationState, handlePageChange } = usePaginationState(location);
+  const { filterState, updateFilterState } = useFilterState(location);
+  if (!journal) return null;
```
Comment (1aurend) [inline client/src/frontend/containers/VolumeDetail/index.js:24]: To Do: same as for `JournalDetail`... use updated hooks and helpers.

File: client/src/hooks/pagination/use-pagination-state/index.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+function getSearch(location) {
+  return queryString.parse(location.search);
+}
+
+function setInitialPaginationState(location) {
+  const { page } = getSearch(location);
+  return {
+    number: page || DEFAULT_PAGE,
+    size: PER_PAGE
```
Comment (1aurend) [inline client/src/hooks/pagination/use-pagination-state/index.js:15]: To Do: change to default and add `perPage` input

File: client/src/hooks/journals/use-select-volume/index.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+const sampleData = fixtures.collectionFactory("volume", 1);
+
+export default function useSelectVolume() {
+  const volume = sampleData;
+  const volumeMeta = {
+    pagination: { totalCount: 12, perPage: 12, currentPage: 1 }
+  };
+  // const issues = volume.relationships.issues
+  return { volume, volumeMeta, issues };
+}
```
Comment (1aurend) [inline client/src/hooks/journals/use-select-volume/index.js:25]: To Do: change to `useSelect` after adding journal dispatch hook

File: client/src/hooks/journals/use-select-journal/index.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+export default function useSelectJournal() {
+  const journal = sampleData[0];
+  console.log(journal);
+  console.log(sampleData);
+  const issuesMeta = {
+    pagination: { totalCount: 12, perPage: 12, currentPage: 1 }
+  };
+  const issues = {};
+  return { journal, issuesMeta, issues };
+}
```
Comment (1aurend) [inline client/src/hooks/journals/use-select-journal/index.js:16]: To Do: change to `useSelect` after adding journal dispatch hook

File: client/src/hooks/filters/use-filter-state/index.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  const updateFilterState = ({ reset, param }) => {
+    if (reset) {
+      setFilterState(setInitialFilterState(location));
+      return;
+    }
+    if (param) {
+      setFilterState(param);
+    }
+  };
+
```
Comment (1aurend) [inline client/src/hooks/filters/use-filter-state/index.js:31]: To Do: check containers to see if any other inputs are needed here

File: client/src/frontend/containers/JournalsList/index.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+  const showPagination = journalsMeta?.pagination?.totalPages > 1;
+
+  return journals ? (
+    <>
+      <CheckFrontendMode debugLabel="ProjectCollections" isProjectSubpage />
+      <RegisterBreadcrumbs
+        breadcrumbs={[
+          { to: lh.link("frontendIssuesList"), label: "Back to issues list" }
+        ]}
+      />
```
Comment (1aurend) [inline client/src/frontend/containers/JournalsList/index.js:28]: To Do: figure out breadcrumbs nav order between these routes

File: client/src/frontend/containers/JournalWrapper/index.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+          }
+        ]}
+      />
+      {childRoutes(route, {
+        childProps: {
+          journal,
+          journalResponse,
+          settings
+          // dispatch,
+          // fetchData
```
Comment (1aurend) [inline client/src/frontend/containers/JournalWrapper/index.js:39]: Question: why does `fetchData` get passed down to the children?

File: client/src/frontend/routes.js [hunk 11]
```diff
@@ -148,93 +148,81 @@ const routes = {
...
+          exact: true,
+          component: "JournalsList",
+          path: "/journals/all",
+          isLibrary: true,
+          helper: (params = {}) => {
+            const query = queryString.stringify(params);
+            if (!query) return "/journals/all";
+            return `/journals/all?${query}`;
+          },
+          routes: []
```
Comment (dananjohnson) [inline client/src/frontend/routes.js:170]: Omit if empty

File: client/src/frontend/routes.js [hunk 12]
```diff
@@ -148,93 +148,81 @@ const routes = {
...
+              helper: (p, n) => `/journals/${p}/volumes/${n}`
+            }
+          ]
+        },
+        {
+          name: "frontendIssue",
+          exact: false,
+          component: "IssueWrapper",
+          path: "/journals/issues/:id",
+          helper: p => `/journals/issues/${p}`,
```
Comment (dananjohnson) [inline client/src/frontend/routes.js:213]: Nit, but `p` seems weird here. Guessing that stands for `project`?

File: client/src/hooks/index.js [hunk 13]
```diff
@@ -32,3 +32,13 @@ export { default as useGrabCurrentTextSection } from "./texts/use-grab-current-t
...
+export { default as useSelectSettings } from "./settings/use-select-settings";
+export { default as useSelectAllIssues } from "./journals/use-select-all-issues";
+export { default as usePaginationState } from "./pagination/use-pagination-state";
+export { default as useFilterState } from "./filters/use-filter-state";
+export { default as useDispatchAllIssues } from "./journals/use-dispatch-all-issues";
+export { default as useSelectAllJournals } from "./journals/use-select-all-journals";
+export { default as useSelectJournal } from "./journals/use-select-journal";
+export { default as useSelectVolume } from "./journals/use-select-volume";
+export { default as useUrlFromState } from "./url/use-url-from-state";
+export { default as useDispatchAllJournals } from "./journals/use-dispatch-all-journals";
```
Comment (dananjohnson) [inline client/src/hooks/index.js:44]: This file is really growing now, so let's keep it easier to parse by grouping these hooks like the ones above. I'd be fine with putting an index file in each subfolder of `hooks` too if you think that's better.

File: client/src/frontend/containers/JournalDetail/index.js (hunk 2 above)
Comment (dananjohnson) [reply client/src/frontend/containers/JournalDetail/index.js:51]: Go ahead and add the new helper to any other containers using this function too.

**Review by dananjohnson — COMMENTED** [review]
@1aurend this is looking great! I really like how you've abstracted a lot of boilerplate stuff between these containers into reusable hooks. I don't see any red flags at the moment, though I wasn't able to view a single journal or issue page. Proceed with your to-dos and I'll take a peek once more before you merge it.

**Pull request closed without merging**